src/function/gds/gds_task.cpp

```diff
@@ -0,0 +1,31 @@
+#include "function/gds/gds_task.h"
+
+#include "graph/graph.h"
+
+namespace kuzu {
+namespace function {
+
+void GDSTask::run() {
+    RangeFrontierMorsel frontierMorsel;
+    auto numActiveNodes = 0u;
```

Comment:
numActiveNodes is not very intuitive in this context.

In Frontiers it is used to indicate how many nodes are valid in current frontier. But here it is more like number of neighbours.

Reply:
I renamed this to numActiveNodesForNextIter. No, this local variable is not counting the number of neighbors. It's counting the number of nodes in the next frontier that has been set active for the next iteration. Basically each thread locally computes the number of nodes in the next frontier they are setting active and aggregate it at the end by calling `sharedState->frontiers.incrementActiveNodesForNextIter(numActiveNodesForNextIter);`.